### PR TITLE
Implemented Noise Radius and Config on Thruster Impulse

### PR DESCRIPTION
**Pull request by shihaocao**
# Implemented Noise Radius and Config on Thruster Impulse

This PR adds a spherical gaussian noise on the thruster impulse. The radius of the noise is specified in a config file.

TODO: do the math to specifiy a good noise radius.

### CODE REVIEW

File: config/parameters/fc/base.txt [hunk 1]
```diff
@@ -5,3 +5,7 @@ fc.leader.fire_time_near   300
 
 fc.follower.fire_time_far     1800
 fc.follower.fire_time_near    300
+
+# Ideally these would be in an actuators folder...
+fc.leader.thruster.noise_radius 1.0
```
Comment (govindchari) [inline config/parameters/fc/base.txt]: renaming this thruster.noise.1sig would be more accurate just so we know it is 1sig

**Comment by govindchari** [comment]
Other than the name of the noise config parameter this pr lgtm

**Comment by govindchari** [comment]
ImpulseVsOnTime_wResiduals

Make the noise 1e-3 Ns 3-sigma. This is conservative as can be seen from the above plot residuals. The assumption that we use is that by being conservative, we do not have to explicitly account for the fact that there are four valves rather that three valves aligned with the coordinate axes of the satellite.

File: config/parameters/fc/base.txt (hunk 1 above)
Comment (shihaocao) [reply config/parameters/fc/base.txt]: will rename!

**Comment by shihaocao** [comment]
I just fixed another bug I think, I normalized the vector before multiplying by the sigma. Otherwise, I think we could potentially be adding more variance than intended?

Unsure if this is how the math works out lol...

File: config/parameters/fc/base.txt (hunk 1 above)
Comment (shihaocao) [reply config/parameters/fc/base.txt]: i think _sigma is sufficient

File: src/psim/fc/orbit_controller.cpp [hunk 2]
```diff
@@ -114,7 +116,18 @@ void OrbitController::step() {
     gnc::control_orbit(_orbit_controller, data, actuation);
 
     if (lin::all(lin::isfinite(actuation.J_ecef))) {
-      J_ecef = actuation.J_ecef;
+      //get random noise
+      Vector3 random_noise = lin::gaussians<Vector3>(_randoms);
+
+      // normalize the random noise
+      random_noise = random_noise / lin::norm(random_noise);
```
Comment (govindchari) [inline src/psim/fc/orbit_controller.cpp]: don't normalize

Comment (shihaocao) [reply src/psim/fc/orbit_controller.cpp]: Lol yeah this is dumb, forgot the extra gaussian anyway

**Review by govindchari — APPROVED** [review]
lgtm

**Pull request merged**